### ISSUE TITLE
Unit tests have warning

### ISSUE THREAD

**Issue by phil-davis** [Behat/Gherkin#145]
e.g. https://travis-ci.org/Behat/Gherkin/jobs/480397897

```
$ vendor/bin/phpunit -v --coverage-clover=coverage.clover
PHPUnit 5.7.27 by Sebastian Bergmann and contributors.
Runtime:       PHP 5.6.32 with Xdebug 2.5.5
Configuration: /home/travis/build/Behat/Gherkin/phpunit.xml.dist
................................................W..............  63 / 190 ( 33%)
............................................................... 126 / 190 ( 66%)
............................................................... 189 / 190 ( 99%)
.                                                               190 / 190 (100%)
Time: 20.44 seconds, Memory: 28.75MB
There was 1 warning:
1) Warning
The data provider specified for Tests\Behat\Gherkin\Keywords\CachedArrayKeywordsTest::testTranslation is invalid.
Outline should have examples table, but got none for outline "Erasing other agents' memory" on line: 35 in file: az_1.feature:
["# language: az\n\u00d6z\u0259llik: Internal operations\n  In order to stay secret\n  As a secret organization\n  We need to be able to erase past agents' memory\n\n  Kontekst:\n    Tutaq ki there is agent A\n    Verilir there is agent A\n    H\u0259m there is agent B\n    V\u0259 there is agent B\n\n  N\u00fcmun\u0259l\u0259r: Erasing agent memory\n    Tutaq ki there is agent J\n    Verilir there is agent J\n    H\u0259m there is agent K\n    V\u0259 there is agent K\n    N\u0259 vaxt ki I erase agent K's memory\n    \u018fg\u0259r I erase agent K's memory\n    O halda there should be agent J\n    Ancaq there should not be agent K\n    Amma there should not be agent K\n\n  Ssenari: Erasing agent memory\n    Tutaq ki there is agent J\n    Verilir there is agent J\n    H\u0259m there is agent K\n    V\u0259 there is agent K\n    N\u0259 vaxt ki I erase agent K's memory\n    \u018fg\u0259r I erase agent K's memory\n    O halda there should be agent J\n    Ancaq there should not be agent K\n    Amma there should not be agent K\n\n  Ssenarinin strukturu: Erasing other agents' memory\n    Tutaq ki there is agent <agent1>\n    Verilir there is agent <agent1>\n    H\u0259m there is agent <agent2>\n    V\u0259 there is agent <agent2>\n    N\u0259 vaxt ki I erase agent <agent2>'s memory\n    \u018fg\u0259r I erase agent <agent2>'s memory\n    O halda there should be agent <agent1>\n    Ancaq there should not be agent <agent2>\n    Amma there should not be agent <agent2>\n\n    N\u00fcmun\u0259l\u0259r:\n      | agent1 | agent2 |\n      | D      | M      |"]
WARNINGS!
Tests: 190, Assertions: 562, Warnings: 1.
```

and only runs 190 tests.

But recently there were 260 tests running - e.g. https://travis-ci.org/Behat/Gherkin/jobs/416234102

PR coming...

**Comment by phil-davis** [comment]
Example CI fail on warning with just the first commit: https://travis-ci.org/phil-davis/Gherkin/builds/480450868

**Comment by stof** [comment]
Thanks for catching this. I rewrote these tests so that they behave properly in case of parsing error.

**Comment by phil-davis** [comment]
Fix to translations file is in PR #148 
Then fix/refactoring/enhancement to unit tests is in #147
